src/NzbDrone.Core/Download/Clients/qBittorrent/QBittorrent.cs

```diff
@@ -0,0 +1,248 @@
+                    case "checkingUP": // torrent has finished downloading and is being checked
+                        item.Status = DownloadItemStatus.Completed;
+                        item.RemainingTime = TimeSpan.Zero; // qBittorrent sends eta=8640000 for completed torrents
+                        break;
+
+                    case "stalledDL": // torrent is being downloaded, but no connection were made
+                        item.Status = DownloadItemStatus.Warning;
+                        item.Message = "The download is stalled with no connections";
+                        break;
+
+                    case "downloading": // torrent is being downloaded and data is being transfered
+                    default: // new status in API? default to downloading
+                        item.Status = DownloadItemStatus.Downloading;
+                        break;
+                }
+
+                queueItems.Add(item);
+            }
+
+            return queueItems;
+        }
+
+        public override void RemoveItem(string downloadId, bool deleteData)
+        {
+            _proxy.RemoveTorrent(downloadId, deleteData, Settings);
+        }
+
+        public override DownloadClientStatus GetStatus()
+        {
+            var config = _proxy.GetConfig(Settings);
+
+            var destDir = new OsPath((string)config.GetValueOrDefault("save_path"));
+
+            return new DownloadClientStatus
+            {
+                IsLocalhost = Settings.Host == "127.0.0.1" || Settings.Host == "localhost",
+                OutputRootFolders = new List<OsPath> { _remotePathMappingService.RemapRemoteToLocal(Settings.Host, destDir) }
+            };
+        }
+
+        protected override void Test(List<ValidationFailure> failures)
+        {
+            failures.AddIfNotNull(TestConnection());
+            if (failures.Any()) return;
+            failures.AddIfNotNull(TestGetTorrents());
+        }
+
+        private ValidationFailure TestConnection()
+        {
+            try
+            {
+                var version = _proxy.GetVersion(Settings);
+                if (version < 5)
+                {
+                    // API version 5 introduced the "save_path" property in /query/torrents
+                    return new NzbDroneValidationFailure("Hostname", "Unsupported client version")
+                    {
+                        DetailedDescription = "Please upgrade to qBittorrent version 3.2.4 or higher."
+                    };
+                }
+                else if (version < 6 && Settings.TvCategory.Length > 0)
+                {
+                    // API version 6 introduced support for labels
+                    return new NzbDroneValidationFailure("Category", "Category is not supported")
+                    {
+                        DetailedDescription = "Labels are not supported until qBittorrent version 3.3.0. Please try again with an empty Category."
```

Comment:
@markus101 I did make this change. But on second thought, do you think it's worth keeping the compatibility code at all?